legend-sdlc-test-generation/src/main/java/org/finos/legend/sdlc/test/junit/AbstractTestClassGenerator.java

```diff
@@ -58,7 +64,18 @@ private String generateTestClassPackage(PackageableElement protocolElement)
 
     private String generateTestClassName(PackageableElement protocolElement)
     {
-        return appendJavaIdentifier(new StringBuilder("Test"), protocolElement.name).toString();
+        return appendJavaIdentifier(new StringBuilder(TEST_PREFIX), generateTestName(protocolElement.name)).toString();
+    }
+
+    private String generateTestName(String name)
+    {
+        int maxByteLength = NAME_BYTE_MAX - TEST_PREFIX.getBytes(StandardCharsets.UTF_8).length;
+        if (name.getBytes(StandardCharsets.UTF_8).length > maxByteLength)
+        {
+            String hashCode = String.valueOf(Math.abs(name.hashCode()));
+            return name.substring(0, maxByteLength - hashCode.getBytes(StandardCharsets.UTF_8).length) + hashCode;
+        }
+        return name;
```

Comment:
I think something like this, which will make the renaming a little more uniform, is better.  This also takes care of a problem with handling of unicode characters that require multiple bytes to encode.
```suggestion
        int overage = name.getBytes(StandardCharsets.UTF_8).length - NAME_BYTE_MAX;
        if (overage > 0)
        {
            int newLen = name.length() - overage - 9;
            while (Character.isSurrogate(name.charAt(newLen - 1))
            {
                len--;
            }
            StringBuilder builder = new StringBuilder(newLen + 9);
            builder.append(name, 0, newLen);
            new Formatter(builder).format("_%08x", name.hashCode());
            return builder.toString();
        }
        return name;
```